workflow/Client.go

```diff
@@ -96,7 +96,8 @@ func (c *Client) postObject(obj interface{}, endpoint string, out interface{}) e
 		return resp.ToError()
 	}
 
-	if resp.StatusCode != http.StatusCreated {
+	if resp.StatusCode != http.StatusCreated &&
+	   resp.StatusCode != http.StatusOK {
```

Comment:
When posting dsl for a query, this needs to be ok with just 200 (because nothing gets created with a query)